### PR TITLE
Metrics/gossip data collection

### LINKED ISSUE #10962: Identify any missing metrics for Gossip data collection

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
Explain your changes:
This PR adds the missing metrics for gossip data collection

When you review it, please pay close attention to how I compute the `Block/Snark_work/Transaction` processing time. I am doing it different from the validation time. The computation of validation time is a little weird because it subtracts the expiration_time with the the time it finishes.

Explain how you tested your changes:
*

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #10962

### CODE REVIEW

File: src/lib/mina_metrics/prometheus_metrics/mina_metrics.ml [hunk 1]
```diff
@@ -431,6 +451,20 @@ module Network = struct
...
+      Moving_time_average
+        (struct
+          include Delay_time_spec
+
+          let subsystem = subsystem
+
+          let name = "processing_time"
+
+          let help =
+            "average time, in ms, for blocks to be accepted or rejected"
```
Comment (deepthiskumar) [inline src/lib/mina_metrics/prometheus_metrics/mina_metrics.ml:475]: Can you create two metrics for accepted and rejected resp.?

File: src/lib/bootstrap_controller/bootstrap_controller.ml [hunk 2]
```diff
@@ -149,8 +150,11 @@ let on_transition t ~sender ~root_sync_ledger ~genesis_constants
         with
         | Ok (`Root root, `Best_tip best_tip) ->
             if done_syncing_root root_sync_ledger then return `Ignored
-            else
+            else (
+              t.num_of_root_snarked_ledger_retargeted <-
+                t.num_of_root_snarked_ledger_retargeted + 1 ;
```
Comment (deepthiskumar) [inline src/lib/bootstrap_controller/bootstrap_controller.ml]: Should we do this only in the `New` case in `start_sync_job_with_peer`?

Comment (ghost-not-in-the-shell) [reply src/lib/bootstrap_controller/bootstrap_controller.ml]: Yes. That's better.

File: src/lib/mina_metrics/prometheus_metrics/mina_metrics.ml [hunk 3]
```diff
@@ -472,7 +472,24 @@ module Network = struct
...
+    module Rejection_time =
+      Moving_time_average
+        (struct
+          include Delay_time_spec
+
+          let subsystem = subsystem
+
+          let name = "rejection_time"
+
+          let help = "average time, in ms, for blocks to be rejected"
```
Comment (deepthiskumar) [inline src/lib/mina_metrics/prometheus_metrics/mina_metrics.ml:492]: Similarly, `average time, in ms, for blocks to be rejected after the OCaml process receives`?

**Review by deepthiskumar — APPROVED** [review]
Suggested a description for the processing time metric but otherwise looks good

**Review by nholland94 — CHANGES_REQUESTED** [review]
One issue I see here is how we are tracking these new metrics. Quite a few of these are just gauges that are set to the most recent timing value we collect, which is not very useful to query from prometheus. I've learned it's better to use rolling totals as we can easily craft useful queries against this type of data using the prometheus query language. Some of these metrics can also be tied into our new `O1trace` threading model to integrate it into that system and simplify the timing code a little.

File: src/lib/transition_frontier/transition_frontier.ml [hunk 4]
```diff
@@ -105,25 +105,22 @@ let load_from_persistence_and_start ~logger ~verifier ~consensus_local_state
...
-    | Error `Sync_cannot_be_running ->
-        Error (`Failure "sync job is already running on persistent frontier")
-    | Error (`Failure _) as err ->
-        err
-    | Ok result ->
-        Mina_metrics.(
-          Gauge.set Persistent_database.reading_from_disk_ms
-            Time.(Span.to_ms @@ diff (now ()) start)) ;
-        Ok result
+    O1trace.sync_thread "persistent_frontier_read_from_disk" (fun () ->
```
Comment (nholland94) [inline src/lib/transition_frontier/transition_frontier.ml:108]: This should be a `O1trace.thread` and not a `sync_thread`, as the computation inside is deferred.

**Review by nholland94 — APPROVED** [review]

**Pull request merged**